src/components/views/elements/Flair.js

```diff
@@ -121,14 +122,47 @@ async function getGroupProfileCached(matrixClient, groupId) {
     }
 
     groupProfiles[groupId] = await matrixClient.getGroupProfile(groupId);
+    groupProfiles[groupId].group_id = groupId;
```

Comment:
Ew for leaving this as a raw server response, underscores and all, and then bolting on the group ID.